pkg/database/service.go

```diff
@@ -266,7 +270,66 @@ func (s service) FindExternalDownload(uuid uuid.UUID) (model.ExternalDownload, e
 	return s.repository.FindExternalDownload(uuid)
 }
 
-func (s service) SaveAs(database *model.Database, instance *model.Instance, stack *model.Stack, newName string, format string) (*model.Database, error) {
+func (s service) Save(database *model.Database, instance *model.Instance, stack *model.Stack) error {
+	tmpName := randomString(20)
+	format := getFormat(database)
+	_, err := s.SaveAs(database, instance, stack, tmpName, format, func(saved *model.Database) {
+		err := s.Delete(database.ID)
+		if err != nil {
+			logError(err)
+			return
```

Comment:
I think you are discarding the error since you are not returning it `Save` will return nil

Reply:
Yeah, but that's because it's running in a go routine and it's likely to happen after the request is done

Reply:
So the save (saveAs) request is async. So users don't really know if saving/saving as actually worked right?

Reply:
No, not until the saving has completed... Or failed. There's a todo comment within the logError function stating that we actually should persist the error message somewhere (possible on the database entity)